src/mbgl/renderer/painters/painter_symbol.cpp

```diff
@@ -68,13 +68,11 @@ void Painter::renderSymbol(PaintParameters& parameters,
         auto paintPropertyValues = layer.iconPaintProperties();
 
         SpriteAtlas& atlas = *bucket.spriteAtlas;
-        const bool iconScaled = layout.get<IconSize>().constantOr(1.0) != 1.0 ||
-            frame.pixelRatio != atlas.getPixelRatio() ||
```

Comment:
Yes, this condition was redundant and misleading. `SpriteAtlas` itself does not have any intrinsic pixel ratio; it has a collection of images of possibly heterogenous pixel ratios.